plugins/API/ProcessedReport.php

```diff
@@ -776,15 +777,27 @@ private function handleSimpleDataTable($idSite, $simpleDataTable, $metadataColum
         );
     }
 
-    private function aggregateReportTotalValues($simpleDataTable, $totals)
+    private function aggregateReportTotalValues($simpleDataTable, $metadateColumns, $totals)
     {
-        $metadataTotals = $simpleDataTable->getMetadata('totals');
+        $metadataTotals = $simpleDataTable->getTotalsRow();
 
         if (empty($metadataTotals)) {
             return $totals;
         }
 
-        $simpleTotals = $this->hideShowMetrics($metadataTotals);
+        $simpleTotals = [];
+
+        // remove columns from totals row that are not in metadata
+        foreach ($metadataTotals as $metadataCol => $metadataValue) {
+            if (
+                isset($metadateColumns[$metadataCol])
+                || preg_match('/^goal_[0-9]+_/', $metadataCol)
+            ) {
+                $simpleTotals[$metadataCol] = $metadataValue;
+            }
+        }
```

Comment:
This filtering is also done for metrics displayed in the processed report, so there is no value in displaying total values for not included metrics.
This change caused the most test updates, as it now hides useless columns from a lot test results.